crates/block-producer/src/block_builder/mod.rs

```diff
@@ -19,61 +14,89 @@ use tracing::{debug, info, instrument};
 use crate::{
     batch_builder::batch::TransactionBatch,
     errors::BuildBlockError,
-    mempool::{BatchJobId, Mempool},
+    mempool::Mempool,
     store::{ApplyBlock, DefaultStore, Store},
-    COMPONENT,
+    COMPONENT, SERVER_BLOCK_FREQUENCY,
 };
 
 pub(crate) mod prover;
 
-use self::prover::{block_witness::BlockWitness, BlockProver};
+use self::prover::{block_witness::BlockWitness, BlockProverKernel};
 
-#[cfg(test)]
-mod tests;
+// FIXME: reimplement the tests.
+// #[cfg(test)]
+// mod tests;
 
-// BLOCK BUILDER
+// BLOCK PROVER
 // =================================================================================================
 
-#[async_trait]
-pub trait BlockBuilder: Send + Sync + 'static {
-    /// Receive batches to be included in a block. An empty vector indicates that no batches were
-    /// ready, and that an empty block should be created.
+pub struct BlockProver {
```

Comment:
Why not call this `BlockBuilder`? In my mind, `BlockProver` would be a service that a block builder would use to prove blocks.